src/Sentry.Unity/Integrations/ReleaseHealthIntegration.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using Sentry.Integrations;
+using UnityEngine;
+
+namespace Sentry.Unity.Integrations
+{
+    public class ReleaseHealthIntegration : ISdkIntegration
+    {
+        private readonly ISceneManager _sceneManager;
+        private readonly ITime _time;
+
+        public ReleaseHealthIntegration() : this(SceneManagerAdapter.Instance, TimeAdapter.Instance)
+        {
+        }
+
+        internal ReleaseHealthIntegration(ISceneManager sceneManager, ITime time)
+        {
+            _sceneManager = sceneManager;
+            _time = time;
+        }
+
+        public void Register(IHub hub, SentryOptions options)
+        {
+            options.DiagnosticLogger?.Log(SentryLevel.Debug, "Registering SceneManager integration.");
+
+            _sceneManager.ActiveSceneChanged += CreateEventListener;
+        }
+
+        private void CreateEventListener(SceneAdapter fromScene, SceneAdapter toScene)
+        {
+            var gameListenerObject = new GameObject("SentryListener");
+            gameListenerObject.hideFlags = HideFlags.HideInHierarchy;
+
+            var gameListener = gameListenerObject.AddComponent<GameEventListener>();
+            gameListener.ApplicationEnter += OnApplicationEnter;
+            gameListener.ApplicationExit += OnApplicationExit;
+        }
+
+        private void OnApplicationEnter()
+        {
+            // If a session has been running and not properly closed
```

Comment:
An idea to have this work with the "app mode" session that the .NET SDK will introduce:

We don't restart the session on the first run (given that the SDK will already have initialized one). The goal is to have a session open even before the first event loop cycle in case there's an error, we can already mark that session as failed.